### PR TITLE
RNMT-6236 Changes for MABS 10 (mainly AndroidX migration)

### PR DESCRIPTION
**Pull request by tiagoappereira**
## Description

This PR includes the following changes:
- Migrates to AndroidX (including the MobileECT aar). This is required for the upcoming MABS 10.
- Drops support for apps targeting SDK 28 (Android 9) or below.
- Fixes some warnings among other goodies

## Context

Closes https://outsystemsrd.atlassian.net/browse/RNMT-6236

## Type of changes

- [x] Feature (change which adds functionality)
    - [x] BREAKING CHANGE (existing functionality will not work as expected due to new feature)
- [ ] Fix (change which fixes an issue)
    - [ ] BREAKING CHANGE (existing functionality will not work as expected due to fix)
- [ ] Performance (change which improves performance)
    - [ ] BREAKING CHANGE (existing functionality will not work as expected due to performance improvement)
- [ ] Refactor (non-breaking change that is neither feature, fix nor performance)
- [ ] Style (non-breaking change that only affects formatting and/or white-space)

## Components affected

- [x] Android
- [ ] iOS
- [ ] JavaScript

## Tests
Tested on all supported Android versions (Android 9 until Android 14)

## Checklist

- [x] Pull request title follows the format `RNMT-XXXX <title>`
- [ ] Tests have been created
- [x] Code follows code style of this project
- [x] CHANGELOG.md file is correctly updated
- [ ] Changes require an update to the documentation
	- [ ] Documentation has been updated accordingly

### CODE REVIEW

File: src/android/osappfeedback.gradle [hunk 1]
```diff
@@ -1,13 +1,13 @@
 repositories{    
-  jcenter()
```
Comment (cmfsotelo) [inline src/android/osappfeedback.gradle:2]: ❤️

**Review by andredestro — DISMISSED** [review]

File: src/android/osappfeedback.gradle [hunk 2]
```diff
@@ -1,5 +1,5 @@
 repositories{    
-  jcenter()
+  mavenCentral()
```
Comment (EiyuuZack) [inline src/android/osappfeedback.gradle]: Should google repo also be added here?

File: src/android/OSAppFeedback.java [hunk 3]
```diff
@@ -74,47 +73,42 @@ protected void setEctViewGroup(ViewGroup ectViewGroup) {
     }
 
 
+    @SuppressLint("ResourceType")
```
Comment (EiyuuZack) [inline src/android/OSAppFeedback.java:61]: What exactly is triggering this warning?

**Review by EiyuuZack — DISMISSED** [review]

File: src/android/OSAppFeedback.java (hunk 3 above)
Comment (tiagoappereira) [reply src/android/OSAppFeedback.java:61]: This [line](https://github.com/OutSystems/Cordova-OutSystems-AppFeedback/pull/22/commits/b83b7e58f6e27629ba6baa293ee6cc75ef6fb4f1#diff-21eef6d954201d941e5303407a6447125c5b0eb7540a75a517604fa02bbb67d8R98).

> Expected resource of type id

File: src/android/osappfeedback.gradle (hunk 2 above)
Comment (tiagoappereira) [reply src/android/osappfeedback.gradle]: Yup, it makes sense! I will add it in a new commit

**Review by cmfsotelo — DISMISSED** [review]
This will effectively break AppFeedback for those sneaky customers that change the minSdk.
Just a stance, not something I think we should revise, hence the approve.

**Comment by tiagoappereira** [comment]
> This will effectively break AppFeedback for those sneaky customers that change the minSdk. Just a stance, not something I think we should revise, hence the approve.

Are you mentioning the drop of `activity.getApplicationInfo().targetSdkVersion >= 29` condition? If yes, it affects the ones changing the targetSdk and not the minSdk. And that, IMO, has way less impact  since I'm not expecting customers to be changing the targetSDK (won't that break the builds also?). But your point still remains!

**Comment by cmfsotelo** [comment]
> > This will effectively break AppFeedback for those sneaky customers that change the minSdk. Just a stance, not something I think we should revise, hence the approve.
> 
> Are you mentioning the drop of `activity.getApplicationInfo().targetSdkVersion >= 29` condition? If yes, it affects the ones changing the targetSdk and not the minSdk. And that, IMO, has way less impact since I'm not expecting customers to be changing the targetSDK (won't that break the builds also?). But your point still remains!

true, my bad!

**Review by cmfsotelo — APPROVED** [review]

**Review by andredestro — APPROVED** [review]

**Review by EiyuuZack — APPROVED** [review]

**Pull request merged**